pkg/network/ovn_kubernetes_dpu_host_test.go

```diff
@@ -182,3 +187,184 @@ func createTestRenderData(ovnNodeMode string) render.RenderData {
+			dpuModeLabel:         "network.operator.openshift.io/dpu=true",
+			expectedOperatorType: corev1.NodeSelectorOpIn,
+			expectedValue:        "true",
+		},
+		{
+			name:                 "smart-nic mode with key+value labels",
+			ovnNodeMode:          "smart-nic",
+			dpuHostModeLabel:     "network.operator.openshift.io/dpu-host=true",
+			smartNicModeLabel:    "network.operator.openshift.io/smart-nic=true",
+			dpuModeLabel:         "network.operator.openshift.io/dpu=true",
+			expectedOperatorType: corev1.NodeSelectorOpIn,
+			expectedValue:        "true",
+		},
+		{
+			name:                 "dpu mode with key+value labels",
+			ovnNodeMode:          "dpu",
+			dpuHostModeLabel:     "network.operator.openshift.io/dpu-host=true",
+			smartNicModeLabel:    "network.operator.openshift.io/smart-nic=true",
+			dpuModeLabel:         "network.operator.openshift.io/dpu=true",
+			expectedOperatorType: corev1.NodeSelectorOpDoesNotExist,
+			expectedValue:        "",
+		},
+		{
+			name:                 "dpu-host mode with key only labels",
+			ovnNodeMode:          "dpu-host",
+			dpuHostModeLabel:     "network.operator.openshift.io/dpu-host=",
+			smartNicModeLabel:    "network.operator.openshift.io/smart-nic=",
+			dpuModeLabel:         "network.operator.openshift.io/dpu=",
+			expectedOperatorType: corev1.NodeSelectorOpExists,
+			expectedValue:        "",
+		},
+		{
+			name:                 "smart-nic mode with key only labels",
+			ovnNodeMode:          "smart-nic",
+			dpuHostModeLabel:     "network.operator.openshift.io/dpu-host=",
+			smartNicModeLabel:    "network.operator.openshift.io/smart-nic=",
+			dpuModeLabel:         "network.operator.openshift.io/dpu=",
+			expectedOperatorType: corev1.NodeSelectorOpExists,
+			expectedValue:        "",
+		},
+		{
+			name:                 "dpu mode with key labels",
+			ovnNodeMode:          "dpu",
+			dpuHostModeLabel:     "network.operator.openshift.io/dpu-host=true",
+			smartNicModeLabel:    "network.operator.openshift.io/smart-nic=true",
+			dpuModeLabel:         "network.operator.openshift.io/dpu=true",
+			expectedOperatorType: corev1.NodeSelectorOpDoesNotExist,
+			expectedValue:        "",
+		},
+	}
+
+	for _, template := range templates {
+		for _, tc := range testCases {
+			testName := template.name + "_" + tc.name
+			t.Run(testName, func(t *testing.T) {
+				g := NewGomegaWithT(t)
+
+				// Create render data
+				data := createTestRenderData(tc.ovnNodeMode)
+				data.Data["DpuHostModeLabel"] = tc.dpuHostModeLabel
+				data.Data["SmartNicModeLabel"] = tc.smartNicModeLabel
+				data.Data["DpuModeLabel"] = tc.dpuModeLabel
+
+				// Extract values from labels for template data
+				extractLabelValues := func(label string) []string {
+					if label == "" {
+						return []string{}
+					}
+					parts := strings.SplitN(label, "=", 2)
+					if len(parts) == 2 && parts[1] != "" {
+						return []string{parts[1]}
+					}
+					return []string{}
+				}
```

Comment:
I think it would be better to use `getKeyValueFromLabel()` function to get the label values rather than writing another function here. It will also test the correctness of the function `getKeyValueFromLabel()`.

Reply:
I agree, thanks for this comment. I changed it.